### PR TITLE
Implemented GitHub actions for CI

### LINKED ISSUE #558: [Feature] Implement Github Actions
As part of the effort to switch from Travis to GH actions, I'm raising this & will begin working on this feature.

### PR DESCRIPTION
**Pull request by marksweb**
This switches to github actions for CI which runs;

- gulp
- flake8
- isort
- unit tests

This resolves #558

### CODE REVIEW

**Review by vinitkumar — APPROVED** [review]
Looks good to me. Let's get this merged and released 🚢  🚀

**Comment by marksweb** [comment]
> Looks good to me. Let's get this merged and released 🚢  🚀

Missed this before I switched it over to coverage. So you'll have missed that commit 👍

File: tests/requirements/base.txt [hunk 1]
```diff
@@ -3,14 +3,15 @@ django-filer>=1.4.0
 djangocms-picture>=2.1.0
 djangocms-link>=2.2.1
 django-polymorphic>=2.0.3
-django-treebeard<4.5
 Pillow
 html5lib>=0.999999999
 
 # other requirements
+coverage
```
Comment (vinitkumar) [inline tests/requirements/base.txt:10]: are we sure to run a unpinned version of packages>

**Review by vinitkumar — CHANGES_REQUESTED** [review]

File: tests/requirements/base.txt (hunk 1 above)
Comment (marksweb) [reply tests/requirements/base.txt:10]: For some things, absolutely.

Coverage is the thing running the test suite, so if there's an issue with it, we'd know at the point code is pushed.

It's also only a package that's involved with the dev side of things. 

There's enough to keep track of with versions of packages that manage functionality that certainly need some degree of pinning. Things like coverage are safe imo

**Review by vinitkumar — APPROVED** [review]

**Comment by marksweb** [comment]
@victor-yunenko Hi, I think you should be able to merge this & remove that requirement for travis to report (given than it's been turned off here, we'll be waiting a while!)

**Pull request merged**